### PR TITLE
add benchmark scripts (in CI as well)

### PR DESCRIPTION
**Pull request by staltz**
Some of this was copied from jitdb. I'm interested in adding this now, before I attempt one trick to speed up initial indexing.

### CODE REVIEW

**Comment by arj03** [comment]
Looks OK. I noticed the scripts in jitdb have a high variance, might be good to run these things with something that averages over multiple runs. Can't remember the package now.

**Comment by staltz** [comment]
Yeah, in CI we don't need to worry too much about time, it can take even many minutes to complete. 

Note I used the same fixture size for both JITDB and ssb-db2. This means that the durations in JITDB are around half a second each test, but here the initial indexing is 9 seconds, it gives you a sense of scale that initial (base index) indexing (with decrypt) is many times heavier than JITDB indexing.

**Review by arj03 — APPROVED** [review]

**Comment by staltz** [comment]
@arj03 I made a new commit, can you review it?

File: benchmark/index.js [hunk 1]
```diff
@@ -42,16 +44,44 @@ if (!skipCreate) {
...
+
+  pull(
+    fromEvent('ssb:db2:migrate:progress', sbot),
+    pull.filter((progress) => progress === 1),
+    pull.take(1),
+    pull.drain(() => {
+      const duration = Date.now() - start
+      t.pass(`duration: ${duration}ms`)
+      fs.appendFileSync(reportPath, `- Migration: ${duration}ms\n`)
+      // wait for new log FS writes to finalize
```
Comment (arj03) [inline benchmark/index.js:70]: Shouldn't a onDrain work here?

**Comment by arj03** [comment]
It's good to see this split into migrate and indexing

File: benchmark/index.js (hunk 1 above)
Comment (staltz) [reply benchmark/index.js:70]: Oh, going to try that.

Comment (staltz) [reply benchmark/index.js:70]: Oh, the problem here is that I don't have `sbot.db`, so I don't have access to the `log` :(

Comment (arj03) [reply benchmark/index.js:70]: Ah right, okay.

Comment (arj03) [reply benchmark/index.js:70]: Feel free to merge :)

**Pull request merged**